### PR TITLE
feat: add deployment of nvidia nim

### PR DESCRIPTION
**Pull request by laurentgrangeau**
Adding deployment of NVIDIA NIMs on the ML platform

### CODE REVIEW

File: best-practices/ml-platform/terraform/features/nim/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+output "inference-url" {
+  value = ""
```
Comment (arueth) [inline best-practices/ml-platform/terraform/features/nim/outputs.tf:2]: If this isn't going to be populated with a value, can you remove it?

**Comment by arueth** [comment]
For the repo, we try and follow the [Terraform on Google Cloud: Best practices for general style and structure](https://cloud.google.com/docs/terraform/best-practices/general-style-structure), could you take a look at it and update the PR. Mainly around the [Adopt a naming convention](https://cloud.google.com/docs/terraform/best-practices/general-style-structure#naming-convention) and using underscore to delimit multiple words.

Although not required, it is good practice to break out standard attributes and block attributes as well as alphabetize attributes in resources and data sources to make merges and updates from multiple people cleaner. 

Also please ensure that you run `terraform fmt` on the code as this will be enforce when merging to main.

**Review by arueth — COMMENTED** [review]
Also see my prior comment.

**Comment by roberthbailey** [comment]
How different is this from the nim guide that we merged into https://github.com/GoogleCloudPlatform/ai-on-gke/tree/main/tutorials-and-examples/nvidia-nim?

**Comment by arueth** [comment]
Looks to be similar, except done with Terraform. We are consuming this in the `ml-platform`, so it will be referenced internally primarily.

**Pull request closed without merging**